components/concurrency_manager/tests/memory_usage.rs

```diff
@@ -0,0 +1,64 @@
+use concurrency_manager::ConcurrencyManager;
+use futures::executor::block_on;
+use rand::prelude::*;
+use std::{mem::forget, mem::ManuallyDrop, thread};
+use txn_types::{Key, Lock, LockType};
+
+// This test is heavy so we shouldn't run it daily.
+// Run it with the following command (recommending release mode) and see the printed stats:
+//
+// ```
+// cargo test --package concurrency_manager --test memory_usage --features jemalloc --release -- test_memory_usage --exact --ignored --nocapture
+// ```
+#[test]
+#[ignore]
+fn test_memory_usage() {
+    // Make key as small as possible, so concurrent writes in the scheduler
+    // can be as many as possible. Then, we can maximize the amplication
+    // caused by the data structure.
+    const KEY_LEN: usize = 16;
+    // We allow 100MB pending writes in the scheduler.
+    const LOCK_COUNT: usize = 100 * 1024 * 1024 / KEY_LEN;
+
+    // Let system collect the memory to avoid drop time at the end of the test.
+    let cm = ManuallyDrop::new(ConcurrencyManager::new(1.into()));
```

Comment:
Does it do anything about how the ConcurrencyManager's content is being dropped?